cask-bootstrap.el

```diff
@@ -54,18 +54,19 @@
             package-archive-contents
             (package-user-dir cask-bootstrap-dir))
         (package-initialize)
-        (condition-case nil
-            (mapc 'require cask-bootstrap-packages)
-          (error
-           (add-to-list 'package-archives (cons "gnu" "https://elpa.gnu.org/packages/"))
-           (add-to-list 'package-archives (cons "melpa" "https://melpa.org/packages/"))
-           (package-refresh-contents)
-           (mapc
-            (lambda (package)
-              (unless (package-installed-p package)
-                (package-install package)))
-            cask-bootstrap-packages)
-           (mapc 'require cask-bootstrap-packages))))
+        (mapc
+         (lambda (package)
+           (condition-case nil
+               (progn
+                 (require package))
+             (error
+              (unless package-archives
```

Comment:
Why check `package-archives` before modifying it?

Reply:
So that `package-archives` modification happens only once, because now `error` handler will be called for every package failed to be `required`.

Reply:
good, but `package-archives` has ELPA entry as init value?
```
package-archives
;;=> (("gnu" . "https://elpa.gnu.org/packages/"))
```

Reply:
It is let-bound to `nil` for this block:

https://github.com/cask/cask/blob/88a89d9d4bb6e17f6ab470c348bd5698a3c2a3b8/cask-bootstrap.el#L52

Reply:
I overlooked, thanks. So, please let bound as you expected in let 1st argument.

Reply:
I am sorry, I am not following. Right now it also serves a second purpose of a `package-initialized-p` variable to ensure we only fetch package contents once. Do you want me to change it?

Reply:
I can't understand what you say "Right now it also serves a second purpose of a package-initialized-p variable to ensure we only fetch package contents once.", please tell me more?
And now your code does `(package-refresh-contents)` multiply in every loops?

Reply:
No, this version of the code does not run refresh multiple times. During the first iteration that needs to install something `package-archives` is `nil`, that block performs the initialization and populates `package-archives` list, so the subsequent iterations never enter the initialization block.

That is what I mean when I say it serves the second purpose of keeping "was initialized" state.

Reply:
@conao3 do you need something else from me here?

Reply:
Sorry late reply, I was very busy recently. Nothing needed, thanks for your contribution!

Reply:
And thank you for your additional description! I finally understood your code is everything ok.

Reply:
Np, thank you for looking after cask!